homeassistant/components/flux_led/light.py

```diff
@@ -171,133 +184,191 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
-
-    def _connect(self):
-        """Connect to Flux light."""
-
-        self._bulb = WifiLedBulb(self._ipaddr, timeout=5)
-        if self._protocol:
-            self._bulb.setProtocol(self._protocol)
-
-        # After bulb object is created the status is updated. We can
-        # now set the correct mode if it was not explicitly defined.
-        if not self._mode:
-            if self._bulb.rgbwcapable:
-                self._mode = MODE_RGBW
-            else:
-                self._mode = MODE_RGB
-
-    def _disconnect(self):
-        """Disconnect from Flux light."""
-        self._bulb = None
+    """Represents a Flux Light entity."""
+
+    def __init__(self, unique_id: str, device: dict, effect_speed: int, bulb):
+        """Initialize the Flux light entity."""
+        self._name = device[CONF_NAME]
+        self._unique_id = unique_id
+        self._icon = "mdi:lightbulb"
+        self._attrs = {}
+        self._state = None
+        self._brightness = None
+        self._hs_color = None
+        self._white_value = None
+        self._current_effect = None
+        self._last_brightness = None
+        self._last_hs_color = None
+        self._ip_address = device[CONF_HOST]
+        self._effect_speed = effect_speed
+        self._mode = None
+        self._get_rgbw = None
+        self._get_rgb = None
+        self._bulb = bulb
+
+    async def async_remove_light(self, device: dict):
+        """Remove a bulb device when it is removed from options."""
+
+        bulb_id = device["device_id"]
+
+        if self._unique_id != bulb_id:
+            return
+
+        entity_registry = await self.hass.helpers.entity_registry.async_get_registry()
+        entity_entry = entity_registry.async_get(self.entity_id)
+
+        device_registry = await self.hass.helpers.device_registry.async_get_registry()
+        device_entry = device_registry.async_get(entity_entry.device_id)
+
+        if (
+            len(
+                async_entries_for_device(
+                    entity_registry,
+                    entity_entry.device_id,
+                    include_disabled_entities=True,
+                )
+            )
+            == 1
+        ):
+            # If only this entity exists on this device, remove the device.
+            device_registry.async_remove_device(device_entry.id)
+
+        entity_registry.async_remove(self.entity_id)
+
+    async def async_added_to_hass(self):
+        """Run when the entity is about to be added to hass."""
+        await super().async_added_to_hass()
+
+        async_dispatcher_connect(
+            self.hass, SIGNAL_REMOVE_DEVICE, self.async_remove_light
+        )
+
+    def update_bulb_info(self):
+        """Update the bulb information."""
+        self._bulb.update_state()
+        self._get_rgbw = self._bulb.getRgbw()
+        self._get_rgb = self._bulb.getRgb()
+
+    @Throttle(timedelta(seconds=DEFAULT_SCAN_INTERVAL))
+    async def async_update(self):
```

Comment:
Should we change to the sync api here too? Then the platform helper will do updates in sequence instead of concurrent.

Reply:
Probably makes sense - that is just `def update(self)` correct?